hydro_deploy/core/src/localhost.rs

```diff
@@ -1,327 +0,0 @@
-use std::collections::HashMap;
-use std::net::SocketAddr;
-#[cfg(unix)]
-use std::os::unix::process::ExitStatusExt;
-use std::path::PathBuf;
-use std::sync::Arc;
-
-use anyhow::Result;
-use async_channel::{Receiver, Sender};
-use async_process::{Command, Stdio};
-use async_trait::async_trait;
-use futures::io::BufReader;
-use futures::{AsyncBufReadExt, AsyncRead, AsyncWriteExt, StreamExt};
-use hydroflow_cli_integration::ServerBindConfig;
-use tokio::sync::RwLock;
-
-use super::{
-    ClientStrategy, Host, HostTargetType, LaunchedBinary, LaunchedHost, ResourceBatch,
-    ResourceResult, ServerStrategy,
-};
-
-struct LaunchedLocalhostBinary {
-    child: RwLock<async_process::Child>,
-    stdin_sender: Sender<String>,
-    stdout_cli_receivers: Arc<RwLock<Vec<Sender<String>>>>,
-    stdout_receivers: Arc<RwLock<Vec<Sender<String>>>>,
-    stderr_receivers: Arc<RwLock<Vec<Sender<String>>>>,
-}
-
-#[async_trait]
-impl LaunchedBinary for LaunchedLocalhostBinary {
-    async fn stdin(&self) -> Sender<String> {
-        self.stdin_sender.clone()
-    }
-
-    async fn cli_stdout(&self) -> Receiver<String> {
-        let mut receivers = self.stdout_cli_receivers.write().await;
-        let (sender, receiver) = async_channel::unbounded::<String>();
-        receivers.push(sender);
-        receiver
-    }
-
-    async fn stdout(&self) -> Receiver<String> {
-        let mut receivers = self.stdout_receivers.write().await;
-        let (sender, receiver) = async_channel::unbounded::<String>();
-        receivers.push(sender);
-        receiver
-    }
-
-    async fn stderr(&self) -> Receiver<String> {
-        let mut receivers = self.stderr_receivers.write().await;
-        let (sender, receiver) = async_channel::unbounded::<String>();
-        receivers.push(sender);
-        receiver
-    }
-
-    async fn exit_code(&self) -> Option<i32> {
-        self.child
-            .write()
-            .await
-            .try_status()
-            .ok()
-            .flatten()
-            .and_then(|c| {
-                #[cfg(unix)]
-                return c.code().or(c.signal());
-                #[cfg(not(unix))]
-                return c.code();
-            })
-    }
-
-    async fn wait(&mut self) -> Option<i32> {
-        let _ = self.child.get_mut().status().await;
-        self.exit_code().await
-    }
-}
-
-struct LaunchedLocalhost {}
-
-type CLIAndMain = (
-    Arc<RwLock<Vec<Sender<String>>>>,
-    Arc<RwLock<Vec<Sender<String>>>>,
-);
-
-pub fn create_broadcast<T: AsyncRead + Send + Unpin + 'static>(
```

Comment:
Sorry for the noisy diff, but this is what you want to compare to `prioritized_broadcast`.